contrib/ci/check-license.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+# Copyright (C) 2021 Mario Limonciello <superm1@gmail.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import glob
+import sys
+import os
+from typing import List
+
+
+def __get_license(fn: str) -> str:
+    with open(fn, "r") as f:
+        for line in f.read().split("\n"):
+            if line.find("SPDX-License-Identifier:") > 0:
+                return line.split(":")[1]
+    return ""
+
+
+def test_files() -> int:
+
+    rc: int = 0
+
+    for fn in glob.glob("**/*.[c|h|py|sh]", recursive=True):
+        if "meson-private" in fn:
+            continue
+        license = __get_license(fn)
+        if not license:
+            print("{} does not specify a license".format(fn))
+            rc = 1
+            continue
+        if not "GPL" in license:
+            print("{} does not contain LGPL or GPL ({})".format(fn, license))
```

Comment:
Are we not LGPL2+ everywhere?

Reply:
Here are the exceptions:
1. EMMC plugin was leveraged from GPL code, I don't think we can just reclassify it.
2. BCM57xx is GPL-2+
3. CPU plugin is GPL-2+
4. Intel SPI plugin is GPL-2+
5. Flashrom is GPL-2